src/neurostatslib/basis.py

```diff
@@ -533,6 +537,218 @@ def _check_n_basis_min(self) -> None:
             )
 
 
+class BSplineBasis(SplineBasis):
+    """
+    B-spline 1-dimensional basis functions.
+
+    Parameters
+    ----------
+    n_basis_funcs :
+        Number of basis functions.
+    order :
+        Order of the splines used in basis functions. Must lie within [1, n_basis_funcs].
+        The B-splines have (order-2) continuous derivatives at each interior knot.
+        The higher this number, the smoother the basis representation will be.
+
+    Attributes
+    ----------
+    order :
+        Spline order.
+
+    Methods
+    -------
+    _evaluate(x_tuple)
```

Comment:
we had decided not to document hidden methods / attributes, right (because we don't want users calling them)? could just replace this with `evaluate` and the information is still right